### PR TITLE
create deployment id

### PR DESCRIPTION
**Pull request by leena-jawale**
This PR create random string for deployment_id

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+```bash
+terraform init
+```
+
+Check the plan
+
+```bash
+terraform plan
+```
+
```
Comment (sharvarikhedkar) [inline README.md:40]: We can also add add  `terraform validate` to validate the configuration before applying it

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+resource "random_string" "deployment_id" {
+  count = "${var.deployment_id != "" ? 0 : 1}"
+
+  length   = 8
+  special  = false
+  upper = false
+  numeric = false
+}
```
Comment (sharvarikhedkar) [inline main.tf]: missing line at EOF

File: output.tf [hunk 3]
```diff
@@ -0,0 +1,3 @@
+output "deployment_id" {
+  value = "${var.deployment_id == "" ? random_string.deployment_id[0].id : var.deployment_id }"
+}
```
Comment (sharvarikhedkar) [inline output.tf]: missing line at EOF

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+variable "deployment_id" {
+  description = "Creates a random string that will be used in tagging for correlating the resources used with a deployment of AAP."
+  type    = string
+}
```
Comment (sharvarikhedkar) [inline variables.tf]: missing line at EOF

File: README.md [hunk 5]
```diff
@@ -1,3 +1,41 @@
...
+These sections will describe required or recommended steps so that your Ansible Automation Platform deployment is as seamless as possible.
+
+
+### Red Hat Enterprise Linux
+
+You will need to use a Red Hat Enterprise Linux (RHEL) Amazon Machine Image (AMI) as the foundation for your deployment.  While this collection will automatically find a public RHEL AMI available from AWS, public images bill for RHEL outside of your subscription for Ansible Automation Platform.
+
+It is recommended that you create a custom AMI that you may then use to deploy RHEL with your subscriptions that come with Ansible Automation Platform.  [Red Hat Image Builder][image-builder] is a utility that makes creating a custom AMI easy.
+
```
Comment (sharvarikhedkar) [inline README.md:16]:
```suggestion


```

File: README.md [hunk 6]
```diff
@@ -1,3 +1,41 @@
...
+```bash
+terraform plan
+```
+
+Apply infrastructure
+
+```bash
+terraform apply
+```
+
```
Comment (sharvarikhedkar) [inline README.md:46]:
```suggestion
```

**Comment by sharvarikhedkar** [comment]
Tested the pr changes, it creates an expected random_string when user does not provide any value, but wondering if  we want to have any validations on deployment_id length because currently its taking any length deployment_id when provided by the user @leena-jawale

**Review by sharvarikhedkar — DISMISSED** [review]
LGTM

File: README.md [hunk 7]
```diff
@@ -1,3 +1,47 @@
...
+Validate configuration
+```bash
+terraform validate
+```
+
+Check the plan
+
+```bash
+terraform plan
```
Comment (jmighion) [inline README.md]:
```suggestion
terraform plan -out
```
idk enough about Terraforms, but this was suggested. Think it'll be worth adding to save the plan?

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,8 @@
+variable "deployment_id" {
+  description = "Creates a random string that will be used in tagging for correlating the resources used with a deployment of AAP."
+  type    = string
+  validation {
+    condition     = length(var.deployment_id) == 8
```
Comment (jmighion) [inline variables.tf]: Should there be more basic conditions? This would allow `!@#$%^&*` which I would guess isn't valid in AWS for an id.

**Review by jmighion — APPROVED** [review]

**Review by sharvarikhedkar — APPROVED** [review]

**Pull request merged**